ambari-server/src/main/java/org/apache/ambari/server/serveraction/upgrades/FixNotebookStorage.java

```diff
@@ -0,0 +1,83 @@
+ * limitations under the License.
+ */
+
+package org.apache.ambari.server.serveraction.upgrades;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+
+import org.apache.ambari.server.AmbariException;
+import org.apache.ambari.server.actionmanager.HostRoleStatus;
+import org.apache.ambari.server.agent.CommandReport;
+import org.apache.ambari.server.orm.entities.RepositoryVersionEntity;
+import org.apache.ambari.server.orm.entities.UpgradeEntity;
+import org.apache.ambari.server.serveraction.AbstractServerAction;
+import org.apache.ambari.server.state.Cluster;
+import org.apache.ambari.server.state.Clusters;
+import org.apache.ambari.server.state.Config;
+import org.apache.ambari.server.utils.VersionUtils;
+
+import com.google.inject.Inject;
+
+/**
+ * During stack upgrade,
+ * update zeppelin.notebook.storage
+ * to org.apache.zeppelin.notebook.repo.FileSystemNotebookRepo
+ * if stack HDP >=2.6.3 and old value org.apache.zeppelin.notebook.repo.VFSNotebookRepo
+ */
+public class FixNotebookStorage extends AbstractServerAction {
+
+  public static final String ZEPPELIN_NOTEBOOK_STORAGE = "zeppelin.notebook.storage";
+  public static final String ZEPPELIN_CONF = "zeppelin-config";
+  public static final String ORG_APACHE_ZEPPELIN_NOTEBOOK_REPO_VFSNOTEBOOK_REPO = "org.apache.zeppelin.notebook.repo.VFSNotebookRepo";
+  public static final String REC_VERSION = "2.6.3.0";
+  public static final String ORG_APACHE_ZEPPELIN_NOTEBOOK_REPO_FILE_SYSTEM_NOTEBOOK_REPO = "org.apache.zeppelin.notebook.repo.FileSystemNotebookRepo";
+
+
+  @Inject
+  private Clusters clusters;
+
+  @Override
+  public CommandReport execute(ConcurrentMap<String, Object> requestSharedDataContext)
+      throws AmbariException, InterruptedException {
+    String clusterName = getExecutionCommand().getClusterName();
+    Cluster cluster = clusters.getCluster(clusterName);
+    UpgradeEntity upgrade = cluster.getUpgradeInProgress();
+    RepositoryVersionEntity repositoryVersionEntity = upgrade.getRepositoryVersion();
+    String targetVersion = repositoryVersionEntity.getVersion();
+
+    Config config = cluster.getDesiredConfigByType(ZEPPELIN_CONF);
+    if (config == null || VersionUtils.compareVersions(targetVersion, REC_VERSION) < 0) {
+      return createCommandReport(0, HostRoleStatus.COMPLETED, "{}",
+          "zeppelin.notebook.storage change not required", "");
+    }
+    Map<String, String> properties = config.getProperties();
+    String oldContent = properties.get(ZEPPELIN_NOTEBOOK_STORAGE);
+    String newContent = ORG_APACHE_ZEPPELIN_NOTEBOOK_REPO_FILE_SYSTEM_NOTEBOOK_REPO;
+
+    if (ORG_APACHE_ZEPPELIN_NOTEBOOK_REPO_VFSNOTEBOOK_REPO.equals(oldContent)) {
+
+      properties.put(ZEPPELIN_NOTEBOOK_STORAGE, newContent);
+      config.setProperties(properties);
+      config.save();
+      return createCommandReport(0, HostRoleStatus.COMPLETED, "{}", "set zeppelin.notebook.storage to org.apache.zeppelin.notebook.repo.FileSystemNotebookRepo", "");
```

Comment:
why do we hard code property names here if we have constant defined? should we use here formatted string instead